vscode-lean4/src/rpc.ts

```diff
@@ -33,13 +33,15 @@ export class Rpc {
     invoke(name: string, args: any[]): Promise<any> {
         this.seqNum += 1
         const seqNum = this.seqNum
-        this.sendMessage({seqNum, name, args})
-        return new Promise((resolve, reject) => this.pending[seqNum] = {resolve, reject})
+        return new Promise((resolve, reject) => {
```

Comment:
I don't think there's a semantic difference here, but ok.

Reply:
I think there's a potential race if `sendMessage` comes back with `messageReceived` immediately and tries to read `this.pending[seqNum]` before the promise has a chance to store `{resolve, reject}`. Putting both in the promise linearizes this so it cannot happen.

Reply:
Ok, I think I understand.  Note that this can only happen if you call messageReceived manually from sendMessage.

Reply:
I thought it could still happen if a `message` event with a response comes in before the `Promise` executes.